From patchwork Mon Oct 22 22:09:11 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Uros Bizjak <ubizjak@gmail.com>
X-Patchwork-Id: 10652601
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A1F7F14DE
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 22 Oct 2018 22:09:25 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 2FD7B28DAD
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 22 Oct 2018 22:09:25 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 23C9229037; Mon, 22 Oct 2018 22:09:25 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C163828DAD
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 22 Oct 2018 22:09:24 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1729492AbeJWG3p (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 23 Oct 2018 02:29:45 -0400
Received: from mail-wm1-f65.google.com ([209.85.128.65]:34465 "EHLO
        mail-wm1-f65.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726800AbeJWG3p (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 23 Oct 2018 02:29:45 -0400
Received: by mail-wm1-f65.google.com with SMTP id z25-v6so10193534wmf.1
        for <kvm@vger.kernel.org>; Mon, 22 Oct 2018 15:09:22 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=AMC8goqovXNPLWk/y+LNMMpktYKvWsjsKglwhEciDPA=;
        b=WL3oPD29+n5zEOOuvJIqI3iw5Nds8fNd1LMW6mR0/Aoah8lzwKWEXdPxzsgtvOxV5l
         +4mjYzYSW2PclRHjqeOdKZWp2p7wMaev3Ajis7S5TFeFiVCrHk72rvbY1CazOlKrf5IK
         k3aOQS0ZaqZdj4hHT+atIMd7O37wBRnzT78V4OobjclAIhmRdnQSQQ6HH0MxfD01OrYG
         xFEWIfd8fXwAgj+lpMjwQ15ZdXm5l+w2AAKgteTioTqRknxX3p/1/uNwtR/BcNStjJfb
         GlfV6QopejUEcRVlAeY00HVr6x8KaA3Z3yW5eQJYnnw2LHiNAjEg/NNcV9b8hbiYBh8U
         whOg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=AMC8goqovXNPLWk/y+LNMMpktYKvWsjsKglwhEciDPA=;
        b=KaKuy/VqJ24DccCcRDzjtl6gNQJtnLridpM6vgxGd15Bz6kRQITZNX5jrkoADYGGZo
         gB7U7VBJUCDhJ4xJFgftgOLp/4C2aTZKDG/6WLvc9PsvK0RRZVQxxmcoWY2cPk+pxnvP
         kpXOV3mr8e+6CXgf6JQ2iX78ELHHvAiUqVR/kN6JrfldIPS5d4qKIokDKMdq1zuQwjDB
         iYEJA8uCn6DpA2xv4oss9gjGlER1e3x3+hRw7IWLBZNO/FWeqF8J9q1/Aaa1SPgEj666
         DZuHddbBfcubBZRJ8zUnyL+xCceOf2SK3DNNG5Dgib3KWcm/VHZ5ebpLQYfP8CNKhBhI
         1oJQ==
X-Gm-Message-State: AGRZ1gLhxHM5jGw+lQ9KcUj+Rk7v/G8R/WamO2KuRE5Tg1bxr7szEQCH
        y+Vmra81BlFVmHhT981+UJwOwzoF
X-Google-Smtp-Source: 
 AJdET5dVo663HwX60E7itPfUFu5tIsS5OqLBxZCH956he88PmC33qNp4AmosUFQTibz7yuOe0m9NpQ==
X-Received: by 2002:a1c:d986:: with SMTP id
 q128-v6mr18751174wmg.110.1540246161021;
        Mon, 22 Oct 2018 15:09:21 -0700 (PDT)
Received: from localhost.localdomain.com (93-103-18-160.static.t-2.net.
 [93.103.18.160])
        by smtp.gmail.com with ESMTPSA id
 c14-v6sm17270841wmh.44.2018.10.22.15.09.19
        (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
        Mon, 22 Oct 2018 15:09:20 -0700 (PDT)
From: Uros Bizjak <ubizjak@gmail.com>
To: kvm@vger.kernel.org, pbonzini@redhat.com
Cc: Uros Bizjak <ubizjak@gmail.com>
Subject: [PATCH] KVM/nVMX: Remove uneeded forward jump in
 nested_vmx_check_vmentry_hw asm
Date: Tue, 23 Oct 2018 00:09:11 +0200
Message-Id: <20181022220911.9340-1-ubizjak@gmail.com>
X-Mailer: git-send-email 2.17.2
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

There is no need to jump just after the jump insn itself. Also, make
code similar to entering guest mode in vmx_vcpu_run.

Signed-off-by: Uros Bizjak <ubizjak@gmail.com>
Reviewed-by: Jim Mattson <jmattson@google.com>
Reviewed-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 arch/x86/kvm/vmx.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 641a65b30685..5600a01fa6d7 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -13240,13 +13240,11 @@ static int __noclone nested_vmx_check_vmentry_hw(struct kvm_vcpu *vcpu)
 
 		/* Check if vmlaunch of vmresume is needed */
 		"cmpl $0, %c[launched](%0)\n\t"
-		"je 1f\n\t"
-		__ex("vmresume") "\n\t"
-		"jmp 2f\n\t"
-		"1: " __ex("vmlaunch") "\n\t"
+		"jne 1f\n\t"
+		__ex("vmlaunch") "\n\t"
 		"jmp 2f\n\t"
+		"1: " __ex("vmresume") "\n\t"
 		"2: "
-
 		/* Set vmx->fail accordingly */
 		"setbe %c[fail](%0)\n\t"
 
